ocs/api_client.py

```diff
@@ -0,0 +1,255 @@
+            client. If its oc-cli then extra work needs to be done in the
+            specific implementation.
+        """
+
+        raise NotImplementedError("get_labels method is not implemented")
+
+
+# All openshift REST client specific imports here
+from oc import openshift_ops
+from openshift.dynamic import exceptions
+
+
+class OCRESTClient(APIClientBase):
+    """ All activities using openshift REST client"""
+
+    def __init__(self):
+        """TODO: get REST client instance from ctx which is shared globally"""
+        self.rest_client = openshift_ops.OCP()
+
+    @property
+    def name(self):
+        return "OCRESTClient"
+
+    def get_pods(self, **kwargs):
+        """
+        Get pods in specific namespace or across oc cluster
+
+        Args:
+            **kwargs: ex: namespace=rook-ceph, label_selector='x==y'
+
+        Returns:
+            list: of pods names,if no namespace provided then this function
+                returns all pods across openshift cluster
+        """
+
+        resource = self.rest_client.v1_pods
+
+        try:
+            kwargs.update({'resource': resource})
+            pod_data = self.api_get(**kwargs)
+        except exceptions.NotFoundError:
+            logger.error("Failed to get pods: resource not found")
+            raise
+        except Exception:
+            logger.error("Unexpected error")
+            raise
+
+        return [item.metadata.name for item in pod_data.items]
+
+    def get_labels(self, pod_name, pod_namespace):
+        """
+        Get labels from a specific pod
+
+        Args:
+            pod_name (str): Name of the pod
+            pod_namespace (str): namespace where this pod lives
+
+        Raises:
+            NotFoundError: If resource not found
+
+        Returns:
+            dict: All the labels on a pod
+        """
+
+        resource = self.rest_client.v1_pods.status
+        try:
+            pod_meta = self.api_get(
+                resource=resource,
+                name=pod_name,
+                namespace=pod_namespace,
+            )
+        except exceptions.NotFoundError:
+            logger.error("Failed to get pods: resource not found")
+            raise
+        except Exception:
+            logger.error("Unexpected error")
+            raise
+
+        data = pod_meta['metadata']['labels']
+        pod_labels = {k: v for k, v in data.items()}
+        return pod_labels
+
+    def create_service(self, **kw):
+        """
+        Args:
+            kw: ex: body={body} for the request which has service spec
+
+        Returns:
+            ResourceInstance
+        """
+        if 'body' not in kw:
+            logger.error("create must have body ")
+
+        resource = self.rest_client.v1_services
+        kw.update({'resource': resource})
+        return self.api_create(**kw)
+
+    def api_get(self, **kw):
+        return self.rest_client.call_api("GET", **kw)
+
+    def api_post(self, **kw):
+        pass
```

Comment:
super(). api_post(**kw) - this way we will raise NotImplementedError as you have defined in abstract method.  The same below.

Reply:
Done